deployment/aerie_db_migration.py

```diff
@@ -0,0 +1,301 @@
+
+  # Add arguments
+  exclusive_args.add_argument(
+    '-a', '--apply',
+    help="apply migration steps to specified databases",
+    action='store_true')
+
+  exclusive_args.add_argument(
+    '-r', '--revert',
+    help="revert migration steps to specified databases",
+    action='store_true')
+
+  parser.add_argument(
+    '--all',
+    help="apply[revert] ALL unapplied[applied] migration steps to all databases if none are provided",
+    action='store_true')
+
+  parser.add_argument(
+    '-db', '--db-names',
+    help="list of databases to migrate. migrates all if unspecified",
+    nargs='+',
+    default=[])
+
+  parser.add_argument(
+    '-p', '--hasura-path',
+    help="the path to the directory containing the config.yaml for Aerie. defaults to ./hasura")
+
+  parser.add_argument(
+    '-e', '--env-path',
+    help="the path to the .env file used to deploy aerie. must define AERIE_USERNAME and AERIE_PASSWORD. defaults to .env",
+    default='.env')
+
+  parser.add_argument(
+    '-n', '--network-location',
+    help="the network location of the database. defaults to localhost",
+    default='localhost')
+
+  # Generate arguments
+  args = parser.parse_args()
+
+  HASURA_PATH = "./hasura"
+  if args.hasura_path:
+    HASURA_PATH = args.hasura_path
+  MIGRATION_PATH = HASURA_PATH+"/migrations/"
+
+  # find all migration folders for each Aerie database
+  migration_db = []
+  to_migrate_set = set(args.db_names)
+  dbs_specified = True
+  if not to_migrate_set:
+    dbs_specified = False
+
+  try:
+    os.listdir(MIGRATION_PATH)
+  except FileNotFoundError as fne:
+    print("\033[91mError\033[0m:"+ str(fne).split("]")[1])
+    sys.exit(1)
+  for db in os.listdir(MIGRATION_PATH):
+    #ignore hidden folders
+    if db.startswith('.'):
+      continue
+    # Only process if the folder is on the list of databases or if we don't have a list of databases
+    if not dbs_specified or db in to_migrate_set:
+      migration = DB_Migration(db)
+      for root,dirs,files in os.walk(MIGRATION_PATH+db):
+        if dirs:
+          migration.add_migration_step(dirs)
+      if len(migration.steps) > 0:
+        # If reverting, reverse the list
+        if args.revert:
+          migration.steps.reverse()
+        migration_db.append(migration)
+        to_migrate_set.discard(db)
+
+  if to_migrate_set:
+    print("\033[91mError\033[0m: The following Database(s) do not contain migrations:\n\t"
+          +"\n\t".join(to_migrate_set))
+    sys.exit(1)
+
+  if not migration_db:
+    print("\033[91mError\033[0m: No database migrations found.")
+    sys.exit(1)
+
+  # Check that hasura cli is installed
+  if not shutil.which('hasura'):
+    sys.exit(f'Hasura CLI is not installed. Exiting...')
+  else:
+    os.system('hasura version')
+
+  # Get the Username/Password
+  username = ""
+  password = ""
+  usernameFound = False
+  passwordFound = False
+  with open(args.env_path) as envFile:
```

Comment:
Not for this PR, but just noting we should probably use something like [python-dotenv](https://github.com/theskumar/python-dotenv) here so we don't have to parse the environment manually.